packages/components/text/src/text.spec.js

```diff
@@ -315,4 +315,21 @@ describe('<Detail>', () => {
       );
     });
   });
+
+  describe('when aria-labelledby attribute is provided', () => {
+    it('should use the attribute and link the text to the label', () => {
+      const labelId = 'text-detail-label';
+      const labelText = 'Number of projects:';
+      const textContent = '23';
+
+      render(
+        <div>
+          <span id={labelId}>{labelText}</span>
+          <Text.Detail aria-labelledby={labelId}>{textContent}</Text.Detail>
+        </div>
+      );
+
+      expect(screen.getByLabelText(labelText)).toHaveTextContent(textContent);
```

Comment:
Thanks @kark 👍 

There was an issue when not providing the `as` property to the `Text.Headline` component which is a required property in typescript but someone could use from a javascript file.
Most probably users will always use the `as` property because, if you don't provide it, you would get an unstyled text, but it's always good to cover all the angles.

Updated here: 95eb8dead565645c031b9f1fcc4ba8ca00a3c87e